website/src/pages/login.page.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { FormEvent, useContext, useState } from "react"
+import { centeredColumn } from "src/sprinkles.css"
+import { centeredForm, loginFormBox, loginHeader, positionButton, skinnyWidth, submitButton } from "./login.css"
+import Layout from "../layout"
+import Link from "src/components/link"
+import { UserContext } from "src/auth"
+
+import {
+  CognitoUserPool,
+  CognitoUser,
+  AuthenticationDetails
+} from 'amazon-cognito-identity-js'
+
+const userPool = new CognitoUserPool({
+  UserPoolId: process.env["DAILP_USER_POOL"] ?? "",
+  ClientId: process.env["DAILP_USER_POOL_CLIENT"] ?? "",
+});
+
+const LoginPage = () => {
+
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  // gets the user and its setUser from the context, to update info on the current user
+  const context = useContext(UserContext);
+
+  function submitLogin(e: FormEvent<HTMLFormElement>, username: string, password: string) {
```

Comment:
I would be inclined to move this function up to the provider and have it provide a simple `logIn` function that takes username and password, which you call here. That way, all code that calls the cognito API lives in a single place.

Reply:
I thought I'd just comment these quick changes I made to make sure I understood: I added login and logout functions to the context type definition, like so: 
<img width="609" alt="image" src="https://user-images.githubusercontent.com/40529597/174151403-d243c661-666d-4b40-ab92-ddf96f4db5f6.png">

Those functions are defined in the UserProvider, and then I created hooks to access those functions: 
<img width="510" alt="image" src="https://user-images.githubusercontent.com/40529597/174151450-0b9e6358-8d62-4a91-a6c8-e40237579a4d.png">

Reply:
Hmm I think the `login` and `logout` functions don't need the first `event` argument since they could be called from any kind of event generally.

Reply:
Removed that event argument!